packages/automation-contracts/scheduler/contracts/VestingSchedulerV2.sol

```diff
@@ -77,79 +79,77 @@ contract VestingSchedulerV2 is IVestingSchedulerV2, SuperAppBase {
     ) private returns (bytes memory newCtx) {
-        newCtx = ctx;
-        address sender = _getSender(ctx);
+        newCtx = params.ctx;
+        address sender = _getSender(params.ctx);
         
         // Default to current block timestamp if no start date is provided.
-        if (startDate == 0) {
-            startDate = uint32(block.timestamp);
+        if (params.startDate == 0) {
+            params.startDate = uint32(block.timestamp);
         }
 
         // Note: Vesting Scheduler V2 doesn't allow start date to be in the past.
         // V1 did but didn't allow cliff and flow to be in the past though.
-        if (startDate < block.timestamp) revert TimeWindowInvalid();
+        if (params.startDate < block.timestamp) revert TimeWindowInvalid();
 
-        if (receiver == address(0) || receiver == sender) revert AccountInvalid();
-        if (address(superToken) == address(0)) revert ZeroAddress();
-        if (flowRate <= 0) revert FlowRateInvalid();
-        if (cliffDate != 0 && startDate > cliffDate) revert TimeWindowInvalid();
-        if (cliffDate == 0 && cliffAmount != 0) revert CliffInvalid();
+        if (params.receiver == address(0) || params.receiver == sender) revert AccountInvalid();
+        if (address(params.superToken) == address(0)) revert ZeroAddress();
+        if (params.flowRate <= 0) revert FlowRateInvalid();
+        if (params.cliffDate != 0 && params.startDate > params.cliffDate) revert TimeWindowInvalid();
+        if (params.cliffDate == 0 && params.cliffAmount != 0) revert CliffInvalid();
 
-        uint32 cliffAndFlowDate = cliffDate == 0 ? startDate : cliffDate;
+        uint32 cliffAndFlowDate = params.cliffDate == 0 ? params.startDate : params.cliffDate;
         // Note: Vesting Scheduler V2 allows cliff and flow to be in the schedule creation block, V1 didn't.
         if (cliffAndFlowDate < block.timestamp ||
-            cliffAndFlowDate >= endDate ||
-            cliffAndFlowDate + START_DATE_VALID_AFTER >= endDate - END_DATE_VALID_BEFORE ||
-            endDate - cliffAndFlowDate < MIN_VESTING_DURATION
+            cliffAndFlowDate >= params.endDate ||
+            cliffAndFlowDate + START_DATE_VALID_AFTER >= params.endDate - END_DATE_VALID_BEFORE ||
+            params.endDate - cliffAndFlowDate < MIN_VESTING_DURATION
         ) revert TimeWindowInvalid();
 
-        bytes32 hashConfig = keccak256(abi.encodePacked(superToken, sender, receiver));
+        bytes32 hashConfig = keccak256(abi.encodePacked(params.superToken, sender, params.receiver));
         if (vestingSchedules[hashConfig].endDate != 0) revert ScheduleAlreadyExists();
         vestingSchedules[hashConfig] = VestingSchedule(
             cliffAndFlowDate,
-            endDate,
-            flowRate,
-            cliffAmount,
-            remainderAmount
+            params.endDate,
+            0 /* claimValidityDate */,
+            params.flowRate,
+            false /* isClaimable */,
```

Comment:
It might be a bit wasteful to have both `isClaimable` and `claimValidityDate` stored when you can infer both from just the `claimValidityDate`.

Reply:
agreed. removed `isClaimable` boolean and now checking if the schedule is claimable by reading `claimValidityDate > 0`.
This implies that "regular" schedules have a `claimValidityDate = 0`. as for "claimable" schedules, either the date is set or if the date is not set by the schedule issuer (no specific claim expiry date), it will be automatically assigned to the early end date (i.e. `endDate - END_DATE_VALID_BEFORE`)